### ISSUE TITLE
Tycho 2.7.5 is requiring dependency to junit org.junit [3.8.2,3.9.0)

### ISSUE THREAD

**Issue by jan-pesta** [eclipse-tycho/tycho#2382]
We are using tycho pom-less build and when migrating from 1.4.0 to 2.7.5 we have problem with a build that it is missing dependency.

We have nowhere in project specified version of junit plugin, only in target platform we specify to use version 
`<unit id="org.junit" version="4.12.0.v201504281640"/>` from orbit repository.

Can some one help me to resolve such problem - or give me some hints where to looks for workaround or solution?

[DEBUG] No solution found because the problem is unsatisfiable.: [Unable to satisfy dependency from tycho-extra-1683030192217 0.0.0.1683030192217 to org.eclipse.equinox.p2.iu; org.junit [3.8.2,3.9.0).; Unable to satisfy dependency from tycho-1683030192267 0.0.0.1683030192267 to org.eclipse.equinox.p2.iu; org.junit [3.8.2,3.9.0).; No solution found because the problem is unsatisfiable.]
[INFO] {osgi.os=win32, osgi.ws=win32, org.eclipse.update.install.features=true, osgi.arch=x86_64}
[ERROR] Cannot resolve project dependencies:
[ERROR]   You requested to install 'org.eclipse.equinox.p2.iu; org.junit [3.8.2,3.9.0)' but it could not be found
[ERROR] 
[ERROR] See https://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.

I do not think I am able to reproduce it in some small test case as I do not know what is causing such problem.

Environment used: 
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: c:\Projekty\Apps\JenkinsData\tools\hudson.tasks.Maven_MavenInstallation\Maven_3.6.3\bin\..
Java version: 11.0.2, vendor: AdoptOpenJDK, runtime: c:\Projekty\Bosch\Java\jdk-11.0.2+9
Default locale: en_US, platform encoding: Cp1250
OS name: "windows 10", version: "10.0", arch: "x86", family: "windows"

**Comment by laeubi** [comment]
This usually happens if you use a JUNIT Classpath container with a never eclipse. Also please note that Tycho 3.x is the current latest release.

**Comment by jan-pesta** [comment]
So we could change our test plugin packaing from **eclipse-test-plugin** to only **eclipse-plugin** ?
Thanks

**Comment by laeubi** [comment]
It depends, for specific instructions you should share a minimal example to show the problem.

Beside that, if it is crucial to your business you can find dedicated support options for Tycho here:
https://github.com/eclipse-tycho/tycho#getting-support

**Comment by jan-pesta** [comment]
Yes, I found out that we have specified in several .classfile
`<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/3"/>`
which lead to this error. (some of test projects does not have junit container specified at all)

When I removed all of them, it started working -> solution found.

I am not sure if the solution should be different way that all eclipse-test-plugins will have specified JUNIT_CONTAINER and if it is possible to mixture /3 with other version e.g /4 as we are using junit3 and junit4 in different test projects.
